Pool query cache

dubdromic opened the pull request:
/cc @clintmiller

dubdromic committed:
Add query caches to Apartment connection pools

dubdromic commented on lib/apartment/connection_pool.rb:
```diff
@@ -5,6 +5,10 @@ class ConnectionPool
 
     def_delegators :connection_handler, :pool_exists?, :remove_connection
 
+    def tenant
```
Ignore this. I don't know where it came from.

dubdromic committed:
Fix typo

clintmiller commented on lib/apartment/connection_pool.rb:
```diff
@@ -12,6 +16,7 @@ def use(spec)
       else
         establish_connection(klass, spec)
       end
+      klass.connection.enable_query_cache!
```
By the way, I think this just returns `true`, so this may unintentionally change the return value of `#use`.

dubdromic committed:
Remove old tenant method

dubdromic committed:
Fix initial connection query caches

When #use is first called, establish_connection sets up a new pool class. Since
we were calling klass.connection.enable_query_cache!, the new connection's
queries wouldn't be cached until the next request.  So this fixes that, in
addition to returning the connection object itself.

clintmiller commented on lib/apartment/connection_pool.rb:
```diff
@@ -7,12 +7,13 @@ class ConnectionPool
     def use(spec)
       klass = class_for_database(spec[:database])
-      if pool_exists?(klass.name)
+      connection = if pool_exists?(klass.name)
```
Style: Indentation for the entire conditional.

dubdromic committed:
Fix nil database query cache, style issue
